oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticDocumentMaker.java

```diff
@@ -53,7 +53,8 @@ protected ElasticDocument initDoc() {
 
     @Override
     protected ElasticDocument finalizeDoc(ElasticDocument doc, boolean dirty, boolean facet) {
-        // evaluate path restrictions is enabled by default in elastic. Always index ancestors
+        // Evaluate path restrictions is enabled by default in elastic. Always index ancestors.
+        // When specifically disabled, we will keep indexing it, but the field won't be used at query time
```

Comment:
It's very good to have such documentation!
.... I wonder if we could also document the reason for this behavior... E.g. we believe that at some point, we will deprecate the ability to _disable_ "evaluate path restrictions", or something like that.

And, why do we change the behavior in this patch? It doesn't sound related to the issue (but maybe I'm wrong, in which case we should document this in the issue).

Reply:
Yes it makes sense.

We could make it less silent... e.g. log a warning... but for compatibility reasons, we can't currently can't throw an exception.